examples/erc1155/tests/erc1155.rs

```diff
@@ -2173,3 +2177,303 @@ async fn support_interface(alice: Account) -> eyre::Result<()> {
 
     Ok(())
 }
+
+// ============================================================================
+// Integration Tests: ERC-1155 Pausable Extension
+// ============================================================================
+
+#[e2e::test]
+async fn pauses(alice: Account) -> eyre::Result<()> {
```

Comment:
There's an argument to be made that we should have e2e tests verifying that `pause` reverts when the contract is already paused (similar for `unpause`). This would simply be to verify that the error is propagated correctly.

The same argument goes for `Erc20` and `Erc721` e2e tests too.

Reply:
Could you expand on this?

Reply:
- `pause` reverts if called when a contract is already in "paused" state ([source](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/504b08372a88eeef2a7d4d2388b0b28f718b5f6d/contracts/src/utils/pausable.rs#L90))
- `unpause` reverts if called when a contract is already _not_ in "paused" state ([source](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/utils/pausable.rs#L107))

We have extensive _unit_ tests for both of these functions. We also have "happy flows" for them in each of our `Pausable` contract versions e2e tests. What we do _not_ have are e2e tests for **reverting cases**.

This isn't necessarily a big deal, but we could play it on the safe side, and have these two missing test cases where applicable just to ensure that these two functions work correctly and propagate errors correctly. 
Wdyt?

Reply:
Now I see, I can add them.

Reply:
Added.